### PR TITLE
#262 | Jira Buckets - query thresholds added

### PR DESCRIPTION
**Pull request by mGasiorek998**
# #262 Widget | Add thresholds for Jira query - Added 

## Description
Added two new inputs for warning and error thresholds in Edit Form for Jira Buckets Widget. 
Buckets will be displayed in color:
- white - if no threshold was passed or when thresholds was not added
- yellow - if warning threshold was passed
- red - if error threshold was passed

## Motivation and Context
Visual changes to Jira Bucket Widget will enhance UX. User will have visual confirmation that given bucket is clogged.

## Screenshots
Capture
Capture2

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the [code style](https://github.com/wttech/cogboard/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] Automated functional tests have been added or modified to cover my changes (if applicable)
- [ ] I have updated the documentation accordingly.

---

I hereby agree to the terms of the Cogboard Contributor License Agreement.

### CODE REVIEW

File: cogboard-webapp/src/components/widgets/dialogFields/JiraBucketsInput.js [hunk 1]
```diff
@@ -4,22 +4,37 @@ import { v4 } from 'uuid';
...
 
   const [buckets, setBuckets] = useState(() =>
     (value || []).map(bucket => {
       return {
         id: v4(),
         bucketName: bucket.bucketName,
+        warningThreshold: bucket.warningThreshold
+          ? bucket.warningThreshold
+          : '',
+        errorThreshold: bucket.errorThreshold ? bucket.errorThreshold : '',
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/dialogFields/JiraBucketsInput.js:37]: could be formatted the same

File: cogboard-webapp/src/components/widgets/types/JiraBucketsWidget.js [hunk 2]
```diff
@@ -8,6 +8,16 @@ import TableBody from '@material-ui/core/TableBody';
...
 import { StyledNoItemsInfo } from '../../Widget/styled';
+import { COLORS } from '../../../constants';
+
+const checkThreshold = ({ issueCounts, errorThreshold, warningThreshold }) => {
+  errorThreshold = parseInt(errorThreshold);
+  warningThreshold = parseInt(warningThreshold);
+
+  if (errorThreshold !== 0 && issueCounts >= errorThreshold) return COLORS.RED;
+  else if (warningThreshold !== 0 && issueCounts >= warningThreshold)
+    return COLORS.YELLOW;
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/types/JiraBucketsWidget.js:19]: needs a fix

**Review by szymon-owczarzak — DISMISSED** [review]
Fix 2 issues please

File: cogboard-webapp/src/components/widgets/types/JiraBucketsWidget.js (hunk 2 above)
Comment (mGasiorek998) [reply cogboard-webapp/src/components/widgets/types/JiraBucketsWidget.js]: Fixed in commit 5db5ec2

File: cogboard-webapp/src/components/widgets/dialogFields/JiraBucketsInput.js (hunk 1 above)
Comment (mGasiorek998) [reply cogboard-webapp/src/components/widgets/dialogFields/JiraBucketsInput.js]: Fixed in commit 5db5ec2

**Review by szymon-owczarzak — DISMISSED** [review]

**Review by szymon-owczarzak — APPROVED** [review]

**Pull request merged**